discrimination-ieee754/src/Data/Discrimination/IEEE754.hs

```diff
@@ -0,0 +1,46 @@
+-- | Discrimination of floating-point numbers via their IEEE754 representation.
+--
+-- This technique is described by http://stereopsis.com/radix.html.
+--
+-- This orders -0.0 < 0.0, and imposes unspecified ordering and distinction
+-- between NaNs.
+
+module Data.Discrimination.IEEE754 where
+
+import Data.Binary.IEEE754 (floatToWord, doubleToWord)
+import Data.Bits (xor, setBit, shift)
+import Data.Discrimination (Grouping(..), Group, Sorting(..), Sort)
+import Data.Functor.Contravariant (Contravariant(contramap))
+import Data.Word (Word32, Word64)
+
+-- TODO(awpr): Do these agree with the IEEE754 "total ordering"?
```

Comment:
Resolved the question with judicious use of QuickCheck -- yes, this is IEEE754 total ordering.